### DISCUSSION TITLE
Unifying Policy and Events Specifications

### DISCUSSION THREAD

**Discussion by lummish** [openmobilityfoundation/curb-data-specification#46] [Data Specification]
Currently, the terminology and values used to specify curb policies is different enough from that used to specify events that it is sometimes impossible to assert that an event or sequence of events is compliant with a policy.

We will need to unify the terminology and data types used to define policies and events if we are going to enable compliance assessment in CDS. A few key points of consideration:

1. Policies apply to user classes, but events do not include a user class (the closest analog is `vehicle_type`). Should events report a user class as well as a vehicle type?
2. Policies apply to certain activities (e.g. `parking`, `no parking`), but CDS does not define a means of accessing what each of these activities *is*. For example, what is the difference between `loading` and `parking`? How do we know which applies to a given event/sequence of events?
3. In [this](https://github.com/openmobilityfoundation/curb-data-specification/discussions/26) discussion question, we define some activities that may be reported alongside an event. This notion of an activity (i.e. what function a vehicle is serving to an end user) is distinct from the notion of an activity as defined in the Policy specification (i.e. what action is a vehicle performing relative to the curb). We should consider modifying our terminology here and possibly also extending the policy specification to allow policies to be specified dependent on the event notion of activity.

There are likely other points to consider, but this should be enough to start generating some useful discussion.

**Comment by schnuerle** [comment]
For 1, user classes from Curbs do combine in some way elements from vehicle, propulsion, and purpose in Events. Right now, all the recommended user classes are at least represented in vehicle, propulsion, and purpose. So if you define a user class in Curb you can capture the same named thing in Events now.  Another hitch is user classes are user-definable by definition. Do you think this is good for now or do you have an idea on changes here? 

For 2, based on what we have defined now with policies in Curbs and activities in Events, how do you propose we align these better? Where it gets tricky for me is in the negative but legitimate policy terms, like "no parking" in item 2 above. Event Types (eg, park_start) seem to align closer to Curb activities (eg, parking).
